src/commands/preview/request_payload.rs

```diff
@@ -0,0 +1,55 @@
+use url::Url;
+use uuid::Uuid;
+
+use super::http_method::HTTPMethod;
+
+pub struct RequestPayload {
+    pub method: HTTPMethod,
+    pub https: u8,
+    pub session: String,
+    pub protocol: String,
+    pub domain: String,
+    pub path: String,
+    pub query: String,
+    pub browser_url: String,
+    pub service_url: String,
+    pub body: Option<String>,
+}
+
+impl RequestPayload {
+    pub fn create(method: HTTPMethod, url: Url, body: Option<String>) -> RequestPayload {
+        let session = Uuid::new_v4().to_simple().to_string();
+        let https = if url.scheme() == "https" { 1 } else { 0 };
+
+        let protocol = if https == 1 { "https://" } else { "http://" }.to_string();
```

Comment:
nit: we should be able to just use `url.scheme()` where we use `protocol`, no?

Reply:
I'm going to keep this bit how it is as we use `protocol` and `https` differently.

`https` is used in the cookie to be sent to the workers domain